app/bundles/LeadBundle/Model/CompanyModel.php

```diff
@@ -78,6 +78,7 @@ public function __construct(FieldModel $leadFieldModel, Session $session, EmailV
     public function saveEntity($entity, $unlock = true)
     {
         $this->setEntityDefaultValues($entity, 'company');
+        $this->getCompanyLeadRepository()->updateLeadsPrimaryCompanyName($entity);
```

Comment:
This will also need to be applied to saveEntities for companies edited in batch via the API.